samples/rust/rust_smp_cpu.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Rust SMP CPU manipulation sample.
+
+use kernel::{cpu::Cpu, error::code::*, prelude::*};
+
+module! {
+    type: RustCpu,
+    name: b"rust_smp_cpu",
+    author: b"Rust for Linux Contributors",
+    description: b"Rust SMP CPU sample",
+    license: b"GPL",
+}
+
+struct RustCpu;
+
+impl kernel::Module for RustCpu {
+    fn init(_name: &'static CStr, _module: &'static ThisModule) -> Result<Self> {
+        pr_info!("Rust SMP CPU sample init.\n");
+
+        let guard = Cpu::lock_current();
+        let id = guard.id();
+        pr_info!("Running on CPU #{id}\n");
+
+        let num_possible = Cpu::num_possible();
+        pr_info!("Max possible CPUs: {num_possible}\n");
+
+        if num_possible == 1 {
+            return Err(ENOSPC);
+        }
+
+        guard.call(|| {
+            pr_info!("Running closure on current locked processor #{id}\n");
+        })?;
+
+        // Assume CPUs zero and one are online.
+        let remote = if id == 0 {
+            Cpu::from_id(1)?
+        } else {
+            Cpu::from_id(0)?
+        };
+        let remote_id = remote.call(|| {
+            let guard = Cpu::lock_current();
+            let id = guard.id();
+            pr_info!("Running closure on processor #{id}\n");
+            id
+        })?;
+        pr_info!("Closure ran on remote processor #{remote_id}\n");
+
+        match Cpu::from_id(num_possible) {
+            Err(_) => {}
+            Ok(_) => {
+                pr_info!("Created CPU with bad processor id {num_possible}; aborting!\n");
+                return Err(EINVAL);
+            }
+        };
```

Comment:
Not sure if there was a particular reason you did it this way, but if not, we probably want to showcase the usual way of doing it, especially in places like the samples:

```suggestion
        if let Ok(_) = Cpu::from_id(num_possible) {
            pr_info!("Created CPU with bad processor id {num_possible}; aborting!\n");
            return Err(EINVAL);
        }
```

Reply:
No particular reason -- bad Rust habits from mostly working on personal projects rather than things that see broader review :)